app/scripts/controllers/main.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
```

Comment:
this file has too much stuff in it. I would extract the following:
* I would create a single service that returns a resource The three API calls you make can all be defined on the same resource. Looks at the 'actions' parameter that you can pass to $resource. So in this way your controller would have access to a BitCoinApi and I could do things like BitcoinApi.currencies() or BitcoinApi.rate(code) or BitcoinApi.history(code, to, from)

* I would extract your chart into a directive, if for no other reason than to cut your teeth on directives.